packages/mobile/src/screens/chat-screen/LinkPreview.tsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect, useState } from 'react'
+
+import type { GestureResponderEvent } from 'react-native'
+import { View, Image } from 'react-native'
+
+import { Text, Link } from 'app/components/core'
+import { audiusSdk } from 'app/services/audius-sdk'
+import { makeStyles } from 'app/styles'
+
+import { REACTION_LONGPRESS_DELAY } from './constants'
+
+const useStyles = makeStyles(({ spacing, palette, typography }) => ({
+  root: {
+    backgroundColor: palette.white,
+    borderTopLeftRadius: spacing(3),
+    borderTopRightRadius: spacing(3),
+    overflow: 'hidden'
+  },
+  rootIsLinkPreviewOnly: {
+    borderRadius: spacing(3)
+  },
+  thumbnail: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '100%',
+    maxHeight: spacing(50)
+  },
+  image: {
+    display: 'flex',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%'
+  },
+  domainContainer: {
+    borderBottomColor: palette.neutralLight7,
+    borderBottomWidth: 1,
+    backgroundColor: palette.white
+  },
+  domain: {
+    paddingHorizontal: spacing(4),
+    paddingVertical: spacing(1),
+    fontSize: typography.fontSize.small,
+    lineHeight: typography.fontSize.small * 1.5,
+    color: palette.neutral
+  },
+  textContainer: {
+    display: 'flex',
+    marginTop: spacing(4),
+    marginHorizontal: spacing(4),
+    backgroundColor: palette.white
+  },
+  title: {
+    fontSize: typography.fontSize.medium,
+    lineHeight: typography.fontSize.medium * 1.5,
+    fontFamily: typography.fontByWeight.bold
+  },
+  description: {
+    fontSize: typography.fontSize.medium,
+    lineHeight: typography.fontSize.medium * 1.5,
+    marginBottom: spacing(4)
+  },
+  noDescriptionMarginBottom: {
+    marginBottom: spacing(4)
+  }
+}))
+
+type LinkPreviewProps = {
+  href: string
+  isLinkPreviewOnly: boolean
+  onLongPress: (event: GestureResponderEvent) => void
+}
+
+type UnfurlResponse = {
+  url: string
+  url_type: string
+  site_name: string
+  title: string
+  description: string
+  image: string
+  html: string
+  favicon: string
+}
+
+export const LinkPreview = (props: LinkPreviewProps) => {
+  const styles = useStyles()
+  const { href, isLinkPreviewOnly, onLongPress } = props
+
+  const [metadata, setMetadata] = useState<Partial<UnfurlResponse>>()
+  const domain = metadata?.url ? new URL(metadata?.url).hostname : ''
```

Comment:
super nit: if `metadata?.url` is truthy, then can simply do `new URL(metadata.url).hostname` and not use the question mark

Reply:
Having to use `  const domain = new URL(metadata?.url ?? '').hostname` since metadata is possibly undefined. Which do you think is cleaner? Personally prefer what we have now I think bc it's more obvious, rather than doing `.hostname` on some possibly empty object.

Reply:
Actually `''`' is an invalid input to `URL` too.

Reply:
Deleted a bunch of comments cuz misunderstood what you were talking about